addons/mail/controllers/thread.py

```diff
@@ -150,40 +151,51 @@ def read_subscription_data(self, follower_id):
             ).ids,
         }
 
-    def _prepare_post_data(self, post_data, thread, partner_emails=None, **kwargs):
-        partners = request.env["res.partner"].browse(post_data.pop("partner_ids", []))
+    def _prepare_message_data(self, post_data, *, thread, **kwargs):
+        res = {
+            key: value
+            for key, value in post_data.items()
+            if key in thread._get_allowed_message_params()
+        }
+        if "attachment_ids" in post_data:
+            attachments = request.env["ir.attachment"].browse(
+                [int(aid) for aid in post_data["attachment_ids"]],
+            )
+            if not attachments._has_attachments_ownership(post_data.get("attachment_tokens")):
+                msg = _(
+                    "One or more attachments do not exist, or you do not have the rights to access them.",
+                )
+                raise UserError(msg)
```

Comment:
```suggestion
            raise UserError(
                self.env._(
                    "One or more attachments do not exist, or you do not have the rights to access them."
                )
            )
```
:pick: 

- Only one line saved by the intermediary variable, I wouldn't bother.
- Should use `self.env._` whenever possible.